### DISCUSSION TITLE
Question on HW2

### DISCUSSION THREAD

**Discussion by adamscottc** [williamgilpin/cphy#15] [homework]
Hi Dr. Gilpin, when I took the section of code you wrote to test my class definition, I'm getting a "module object is not callable" error that occurs at the plt.figure() for the model.stored_eigenvalues object. Do you have any tips for how to get rid of this? I looked up the error and apparently it occurs when you try to import a module and then try to call it as a function.

**Comment by williamgilpin** [comment]
hi, is this when you import my code and then run it in your notebook? or is it a class within your code?

**Comment by adamscottc** [comment]
It's a class in my code.

**Comment by williamgilpin** [comment]
Does the following run:

`plt.plot(model.components_, '.')`

If so, then the problem is probably with the `eigmax_numpy` function rather than your code

**Comment by adamscottc** [comment]
The plt.plot function doesn't run, but the eigmax_numpy function does work
in the previous line where I print out the final solution. The
print(model.singular_values_) also works.

**Comment by williamgilpin** [comment]
okay that's weird that the eigs print but the plotting isn't working. It sounds like there might be a () somewhere there shouldn't be, since parentheses trigger a variable being treated like a function. Another option would be that matplotlib is struggling. Does `plt.plot(np.ones(100))` work?

**Comment by adamscottc** [comment]
I guess I just can't tell why Matplotlib wouldn't be working. Substituting
np.ones(100) didn't plot either

**Comment by williamgilpin** [comment]
Okay that's a good sign, that just means it's a matplotlib issue. Just out of curiosity, do you have `plt.plt` anywhere in your code (note that there is no `o` in either)? Otherwise, my guess is that `plt.plot` got reassigned somehow, re-running `import matplotlib.pyplot as plt` might fix it (or just saving and restarting the notebook)

**Comment by adamscottc** [comment]
That worked. Thank you!

**Comment by adamscottc** [comment]
HI Dr. Gilpin! I'm working on problem 2, but Visual Studios isn't recognizing the ipywidgets library. I made sure to install the library using the Anaconda Prompt, but that didn't make a difference. Any advice?

**Reply by williamgilpin** [reply]
Hi, if you installed the library, you might need to re-start your kernel / close-and-reopen the notebook before Jupyter will recognize it

If it's still not working, it's fine to just ignore the code cells requiring the widgets, those cells are just so you can play with the dataset

**Comment by leviyevalex** [comment]
If you would still like to play with the widgets, I'd recommend just opening the notebook with Jupyter, sometimes there are missing plugins, etc... in vscode. It will work natively in Jupyter
